From a413d552ac6509d03847229e4b2c8548bca48ade Mon Sep 17 00:00:00 2001
From: Nandha Kishore Easwaran <quic_nandhaki@quicinc.com>
Date: Thu, 14 Mar 2024 10:37:20 +0530
Subject: [PATCH] wifi: ath12k: Prevent KASAN warning in _find_next_bit

for_each_set_bit() invoking _find_next_bit expects only unsigned long
arguments which causes KASAN warning for u16 variable usage.
Fix this by using ahvif->links_map directly as an argiment to
for_each_set_bit function.

Signed-off-by: Nandha Kishore Easwaran <quic_nandhaki@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/fse.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/fse.c b/drivers/net/wireless/ath/ath12k/fse.c
index ea761fc..3caded7 100644
--- a/drivers/net/wireless/ath/ath12k/fse.c
+++ b/drivers/net/wireless/ath/ath12k/fse.c
@@ -54,7 +54,6 @@ void *ath12k_fse_get_ab_from_netdev(struct net_device *dev,
 	struct ath12k_peer *peer;
 	struct ath12k_vif *ahvif;
 	struct ath12k_link_vif *arvif;
-	u16 links_map;
 	u8 link_id;
 
 	wdev = dev->ieee80211_ptr;
@@ -65,9 +64,7 @@ void *ath12k_fse_get_ab_from_netdev(struct net_device *dev,
 
 	ahvif = ath12k_vif_to_ahvif(vif);
 
-	links_map = ahvif->links_map;
-
-	for_each_set_bit(link_id, &links_map, IEEE80211_MLD_MAX_NUM_LINKS) {
+	for_each_set_bit(link_id, &ahvif->links_map, IEEE80211_MLD_MAX_NUM_LINKS) {
 		arvif = ahvif->link[link_id];
 
 		if (WARN_ON(!arvif))
-- 
2.34.1

